pkg/controller/daemon/daemoncontroller.go

```diff
@@ -994,32 +990,37 @@ func (dsc *DaemonSetsController) syncDaemonSet(key string) error {
 
 	if ds.DeletionTimestamp != nil || !dsc.expectations.SatisfiedExpectations(dsKey) {
 		// Only update status.
-		return dsc.updateDaemonSetStatus(ds, hash)
+		return dsc.updateDaemonSetStatus(ds, hash, nodeList)
 	}
 
-	err = dsc.manage(ds, hash)
+	err = dsc.manage(ds, hash, nodeList)
 	if err != nil {
 		return err
+	} else {
```

Comment:
With this `else` the rest of the code will never be reached

Reply:
@janetkuo @enisoc - PTAL

Reply:
> Assuming that dsc.expectations.SatisfiedExpectations(dsKey) works as expected then if mange adds/delete pods then rollingUpdate will not be called.

When `manage` adds/deletes pods, it first `SetExpectations` and then call `CreationObserved`/`DeletionObserved` when the pod is created/deleted successfully. `SatisfiedExpectations` doesn't mean that there's no pods addition/deletion.